### PR TITLE
[G2M] Table with bar prototype

### LINKED ISSUE #216: create table with bar prototype widget

### PR DESCRIPTION
**Pull request by kc-leung**
fixes #216 

Need: https://github.com/EQWorks/lumen-table/pull/38 

- Added table widget with barColumn display option
- Added multiple unique control options for table widget (  hidePagination, headerTitle, compactTable, centerHeader, barColumns, borderType and headerColor )

Screenshot 2023-02-03 at 4 33 28 PM

Screenshot 2023-02-03 at 4 42 14 PM

### CODE REVIEW

**Review by Hamaad102 — APPROVED** [review]

**Comment by Hamaad102** [comment]
Before merging this @kc-leung can you also bump lumen-labs

**Comment by geoerika** [comment]
> Before merging this @kc-leung can you also bump lumen-labs

@kc-leung leave it, I already did it in my following PR.

File: src/constants/table-widget.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export const tableUniqueOptions = {
```
Comment (geoerika) [inline src/constants/table-widget.js:1]: Look in `constants/types` to refactor this.

File: src/controls/editor-mode/components/color-scheme-controls.js [hunk 2]
```diff
@@ -153,6 +153,11 @@ const ColorSchemeControls = () => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [color, selectedColorIndex, update])
 
+  const getColorType = useMemo(() => {
```
Comment (geoerika) [inline src/controls/editor-mode/components/color-scheme-controls.js:156]: @kc-leung I don't think I understand why you needed to restrict the color type picker for Table. No matter what you choose (HEX, RGB, HSL) the base color returned is always in HEX format.

File: src/controls/editor-mode/right-sidebar.js [hunk 3]
```diff
@@ -114,7 +114,7 @@ const EditorRightSidebar = () => {
       {
         renderRow(null,
           <>
-            {type !== types.MAP &&
+            {![types.Map, types.TABLE].includes(type) &&
```
Comment (geoerika) [inline src/controls/editor-mode/right-sidebar.js:117]: Typo error: it is `types.MAP`

File: src/constants/type-info.js [hunk 4]
```diff
@@ -271,7 +271,43 @@ export default {
     icon: Icons.Table,
     adapter: LocalAdapters[types.TABLE],
     mustGroup: false,
-    uniqueOptions: {},
+    uniqueOptions: {
+      hidePagination: {
+        name: 'Hide Pagination',
```
Comment (geoerika) [inline src/constants/type-info.js:276]: Change name to simple: `Pagination`

File: src/constants/type-info.js [hunk 5]
```diff
@@ -271,7 +271,43 @@ export default {
...
+        type: Boolean,
+        defaultValue: false,
+      },
+      compactTable: {
+        name: 'Compact Table',
+        type: Boolean,
+        defaultValue: false,
+      },
+      centerHeader: {
+        name: 'Center Header',
```
Comment (geoerika) [inline src/constants/type-info.js:291]: Change Name to`Centered Header`. We tend not to use verbs in these names. I just change it to keep consistency.

File: src/controls/editor-mode/components/unique-option-controls.js [hunk 6]
```diff
@@ -6,56 +6,112 @@ import { makeStyles } from '@eqworks/lumen-labs'
...
-                </div>
-              )
-            default:
-              return null
-          }
-        })
-      }
-    </div>
+    <>
+      <div className={classes.uniqueOptionsBooleanContainer}>
```
Comment (geoerika) [inline src/controls/editor-mode/components/unique-option-controls.js:57]: This breaks with the styling for the other widgets. Please check the styling for Display options (specifically the space between the rows of different elements) for the bar widget stories and add styling to match the section there.

The lower part grid for the outlined area should use all the space in the row, each radio should have the same space.
Screen Shot 2023-02-08 at 4 53 23 PM

**Review by geoerika — CHANGES_REQUESTED** [review]
@kc-leung a good start, but I would like the Display Option to be better styled to have it similar to the other widget controls.

**Review by geoerika — APPROVED** [review]
Looks good!

**Review by vlatawiec — APPROVED** [review]

**Pull request merged**